FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/ApiService.kt

```diff
@@ -5,7 +5,6 @@
 
```

Comment:
What's the difference between `ApiService` and `ApiRespository`? Could they not be the same class? There is not really any code in either other than API definitions, looks about the same to me.

Reply:
> What's the difference between ApiService and ApiRespository?

`ApiService` is a part of [retrofit](https://square.github.io/retrofit/). This class describes API calls.
`ApiRespository` is an interface where we describe methods that call API from `ApiService`. It can be used for testing. For example, in tests, we can inject some instance that implements `ApiRespository` but has some settings for testing(like switching to use local domains and etc.)
`FlowcryptApiRepository` it's the realization of `ApiRespository` that we use for the app in production and for debugging.

For now, we don't use an additional realization of `ApiRespository` in tests. This abstraction was created for future needs. If it looks redundant, please let me know and I will delete `ApiRespository` and just leave `FlowcryptApiRepository`.

> Could they not be the same class?

unfortunately no due to the above explanation.

Reply:
Ah, `ApiRespository` is an interface. Why not call it `ApiServiceInterface` (also the file name) ? Then it's obvious from the name. ApiService, ApiRepository, ApiCalls, Api... It all looks the same. Ideally the name should make it clear how is one different from another.

Reply:
I deleted my comment from just now. I'm still struggling to find a proper naming that would make all of this obvious.

Is it possible to rename `ApiService` (the retrofit interface) to anything we want? How about `RetrofitApiDefinitionsInterface` ?

Regarding `ApiRepository` and `FlowcryptApiRepository`, any abstraction that is unused is a maintenance burden. Abstractions are like management layers, too many and nothing can get done. Too few and nothing can get done. So it's important to abstract things when it helps, and refrain from abstracting when it doesn't help. So I would merge them, just call the result `ApiClient`?

Reply:
`...Service` - it's a common approach for `retrofit`. For example

![image](https://user-images.githubusercontent.com/2863246/214600911-f4010aab-c786-46ed-8ea7-44e74e534739.png)

`...Repository`- it's a common [approach](https://github.com/google-developer-training/android-kotlin-fundamentals-starter-apps/tree/master/RepositoryPattern-Starter) in the Android practice

These things are usuall for Android devs. 
I understand your comments. These names can confuse a little that devs who are not familiar with common Android approaches.

Would you like I add `Interface` suffix to these names? Will it look better for understanding?

Reply:
> Is it possible to rename ApiService (the retrofit interface) to anything we want? How about RetrofitApiDefinitionsInterface ?

looks good for me. I will add the change right now.

Reply:
> Regarding ApiRepository and FlowcryptApiRepository, any abstraction that is unused is a maintenance burden. Abstractions are like company management layers, too many and nothing can get done. Too few and nothing can get done. So it's important to abstract things when it helps, and refrain from abstracting when it doesn't help.

I'm not sure about that. It can be used if we will use dependency injection. For now, I'm not going to use it.

> If we don't have any current or planned use for the abstraction, I would merge them, and call the result ApiClient? 

Good one. I like it. I will add changes.

Reply:
> ...Service - it's a common approach for retrofit. For example

> > Is it possible to rename ApiService (the retrofit interface) to anything we want? How about RetrofitApiDefinitionsInterface ?
> 
> looks good for me. I will add the change right now.

Let's do middle ground - `RetrofitApiServiceInterface`

Reply:
> ...Repository- it's a common [approach](https://github.com/google-developer-training/android-kotlin-fundamentals-starter-apps/tree/master/RepositoryPattern-Starter) in the Android practice

Then let's call the result of merging `ApiRepository` and `FlowCryptApiRepository` -> `ApiClientRepository`. That way it could be obvious to both android and non-android context.

Reply:
> Let's do middle ground - RetrofitApiServiceInterface

Will it look informative? I mean use `val apiService` instead of `val retrofitApiServiceInterface`

`val apiService = ApiHelper.getInstance(context).retrofit.create(RetrofitApiServiceInterface::class.java)`

Reply:
> val retrofitApiService would probably be most appropriate, but quite wordy. Do you have to do this a lot?

That's not a problem. IDE does it for me.

> Actually, in which cases do you use retrofitApiService directly and in which cases do you use the ApiClientRepository ? How do you decide which one to use? Or does one use the other? 

`ApiClientRepository` uses `retrofitApiService`. `ApiClientRepository` it's a bridge between the app and APIs.
In one or more places, I use `retrofitApiService` directly. Some logic needs that.

Reply:
Ok. In that case, `retrofitApiService` and `apiClientRepository` sound best.